src/main/java/fr/polytech/sim/log/NamedLogger.java

```diff
@@ -16,4 +16,18 @@ protected NamedLogger(String name) {
         this.name = name;
     }
 
+    /**
+     * Method to build the message put in the logfiles
+     * @param format
+     * @param args
+     * @return
+     */
+    public void log(String format, Object... args){
+        String entry = String.format(format, args);
+        String message = String.format("%s\t%s\n", name, entry);
+        showMessage(message);
+    }
+
+    public abstract void showMessage(String message);
```

Comment:
```suggestion
    protected abstract void showMessage(String message);
```

Aucune raiosn pour que cette méthode soit publique (elle devrait rester `protected` dans les classes filles aussi.